### PR TITLE
Mandatory service name

### PR DESCRIPTION
**Pull request by mattbennett**
Mostly uninteresting except the changes to ``get_service_name`` and the third commit

### CODE REVIEW

File: nameko/containers.py [hunk 1]
```diff
@@ -29,7 +29,12 @@
 
 
 def get_service_name(service_cls):
-    return getattr(service_cls, "name", service_cls.__name__.lower())
+    service_name = getattr(service_cls, 'name', None)
+    if service_name is None:
+        raise RuntimeError('Service class must define a `name` attribute')
```
Comment (davidszotten) [inline nameko/containers.py]: do we want some custom `ConfigurationError`? (e.g. to catch and display something nicer than a traceback in `nameko run`) or are we happy with a traceback in this case?

File: test/test_container.py [hunk 2]
```diff
@@ -527,3 +527,30 @@ def test_stop_during_kill(container, logger):
...
         ]
+
+
+def test_get_service_name():
+
+    class Service():
+        name = "str"
+
+    class UnicodeService():
+        name = u"unicode"
```
Comment (davidszotten) [inline test/test_container.py]: could we have some non-ascii chars in here for good measure?

**Comment by davidszotten** [comment]
at least `index.rst` also needs updating. may want e.g. `hello_service` or some other `foo_service` as the name (as opposed to `hello_world`; c.f. hello world example)

File: nameko/cli/run.py [hunk 3]
```diff
@@ -110,7 +110,13 @@ def _bad_call():
 def run(services, config, backdoor_port=None):
     service_runner = ServiceRunner(config)
     for service_cls in services:
-        service_runner.add_service(service_cls)
+        try:
```
Comment (davidszotten) [inline nameko/cli/run.py]: don't think this is the right place to catch these (`run` may be used from other libraries)

`nameko/cli/main.py` is probably the place (already catches and prints `CommandError`s)

Comment (mattbennett) [inline nameko/cli/run.py]: Quite right, thanks

**Comment by davidszotten** [comment]
:+1:

**Pull request merged**